Model/Observer/RecommendSettings.php

```diff
@@ -0,0 +1,216 @@
+            if (
+                $changedPath == $this->configHelper::IS_RECOMMEND_RELATED_PRODUCTS_ENABLED
+                && $this->configHelper->isRecommendRelatedProductsEnabled()
+            ) {
+                $this->validateRelatedProducts($changedPath);
+            }
+
+            // Validate before enable related products on cart page
+            if (
+                $changedPath == $this->configHelper::IS_RECOMMEND_RELATED_PRODUCTS_ENABLED_ON_CART_PAGE
+                && $this->configHelper->isRecommendRelatedProductsEnabledOnCartPage()
+            ) {
+                $this->validateRelatedProducts($changedPath);
+            }
+
+            // Validate before enable trending items on PDP
+            if (
+                $changedPath == $this->configHelper::IS_TREND_ITEMS_ENABLED_IN_PDP
+                && $this->configHelper->isTrendItemsEnabledInPDP()
+            ) {
+                $this->validateTrendingItems($changedPath);
+            }
+
+            // Validate before enable trending items on cart page
+            if (
+                $changedPath == $this->configHelper::IS_TREND_ITEMS_ENABLED_IN_SHOPPING_CART
+                && $this->configHelper->isTrendItemsEnabledInShoppingCart()
+            ) {
+                $this->validateTrendingItems($changedPath);
+            }
+
+            // Validate before enable looking similar on PDP
+            if (
+                $changedPath == $this->configHelper::IS_LOOKING_SIMILAR_ENABLED_IN_PDP
+                && $this->configHelper->isLookingSimilarEnabledInPDP()
+            ) {
+                $this->validateLookingSimilar($changedPath);
+            }
+
+            // Validate before enable looking similar on cart page
+            if (
+                $changedPath == $this->configHelper::IS_LOOKING_SIMILAR_ENABLED_IN_SHOPPING_CART
+                && $this->configHelper->isLookingSimilarEnabledInShoppingCart()
+            ) {
+                $this->validateLookingSimilar($changedPath);
+            }
+        }
+    }
+
+    /**
+     * @param string $changedPath
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateFrequentlyBroughtTogether(string $changedPath): void
+    {
+        try {
+            $recommendations = $this->recommendManagement->getBoughtTogetherRecommendation($this->getProductId());
+            if (empty($recommendations['renderingContent'])) {
+                $this->configWriter->save($changedPath, 0);
+                throw new LocalizedException(__(
+                    "It appears that there is no trained model available for Frequently Bought Together recommendation for the AppID: %1.",
+                    $this->configHelper->getApplicationID()
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->configWriter->save($changedPath, 0);
+            throw new LocalizedException(__($e->getMessage()));
```

Comment:
Seems to me we should be providing a more informative message in the exception. e.g.
```suggestion
            throw new LocalizedException(__("Unable to save your Recommend configuration due to the following error: " . $e→getMessage()));
```
  WDYT?

Reply:
That's a good point. I'll add this suggested change in a few.